cmd/ipfs/daemon.go

```diff
@@ -423,6 +446,35 @@ func serveHTTPGateway(req cmds.Request) (error, <-chan error) {
 	return nil, errc
 }
 
+type listing struct {
+	Uri  string
+	Keys []string
+}
+
+func loadKeySetFromURLs(urls []string) (key.KeySet, error) {
+	ks := key.NewKeySet()
+	for _, url := range urls {
+		resp, err := http.Get(url)
```

Comment:
woah, no. we should be caching this locally. the http thing should be only to _update_ the list. if list host is unreachable, then what? no gateway? no blocking?